canonical-websites/snapcraft-flask
# Basic skeleton of a Flask app

This just forms the basic skeleton of a Flask app:

- `.gitignore` - Our standard gitignore
- `.travis.yml` - Run flake8 and tests.py
- `requirements.txt` - Python requirements
- `app.py` - The basic application entrypoint
- `tests.py` - Tests with unittest
- `templates/` - Where HTML page templates live
  - `templates/_layout.html` - Basic HTML wrapper
  - `templates/404.html` - 404 page template
- `static/` - For static assets
  - `static/confused-hex.svg` - An image for the 404 page

At the moment it defines no routes, so everything will 404. And it includes one test - to make sure the homepage 404s, as it should for the time being (the homepage lives over at https://github.com/canonical-websites/snapcraft.io).

Fixes #2
Fixes #5
Fixes #7

QA
--

First, set up a python environment:

``` bash
python3 -m venv env3       # Create python3 environment files in a `env3/` directory
source env3/bin/activate  # Update your PATH to use this encapsulated environment
```

Install dependencies:

``` bash
pip install -r requirements.txt
```

Now run the app:

``` bash
FLASK_APP=app.py flask run
```

Visit <http://127.0.0.1:5000>, and check you see a 404 page.

Also, check the tests pass with:

``` bash
./tests.py
```
